src/Cli/Command/CheckCommand.php

```diff
@@ -50,11 +56,31 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $client = $this->getClient();
         $domain = $input->getArgument('domain');
 
+        $solverName = strtolower($input->getOption('solver'));
+        if (!$this->getContainer()->has('challenge_solver.'.$solverName)) {
+            throw new \UnexpectedValueException(sprintf('The solver "%s" does not exists', $solverName));
```

Comment:
Could you specify the list of available solvers in the message (`available: http, dns`)? Better for UX IMO